pkg/operator/ceph/csi/secrets.go

```diff
@@ -212,6 +304,123 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, clusterSpec, clusterNamespaced)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateCephStatusWithCephxStatus updates the cephCluster status with CSI cephxStatus
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, clusterSpec *cephv1.ClusterSpec, name types.NamespacedName) error {
+	keyRotationEnabled, err := keyring.ShouldRotateCephxKeys(clusterSpec.Security.CephX.CSI.CephxConfig, clusterInfo.CephVersion, clusterInfo.CephVersion, cephv1.CephxStatus{})
+	if err != nil {
+		return errors.Wrap(err, "failed to determine if cephx keys should be rotated")
+	}
+	cephCluster := &cephv1.CephCluster{}
+	err = retry.OnError(
+		retry.DefaultRetry,
+		func(err error) bool {
+			return err != nil
+		},
+		func() error {
+			if getErr := context.Client.Get(clusterInfo.Context, name, cephCluster); getErr != nil {
+				logger.Errorf("failed to retrieve cephCluster %q in namespace %q to update csi cephx status. %v", name, name.Namespace, getErr)
+				return getErr
+			}
```

Comment:
In general, I'd suggest looking at the RGW update status method and follow the patterns there. That will at least help keep our update status methods similar across reconcile routines.
https://github.com/rook/rook/blob/a2b0b6449c3c7180a9202d591504fc442a4a6de7/pkg/operator/ceph/object/status.go#L40

Specifically here, `getErr` doesn't have a life outside of this, so `err` is more idiomatic for golang.

Also, we should try to avoid logging error levels widely in the operator. This error is recoverable eventually, so we don't need to worry users too much. Info level would be fine. Additionally, logging this at all isn't necessary inside the loop. It'll retry, and the last error hit will be returned if the retry fails entirely.

```suggestion
			if err := context.Client.Get(clusterInfo.Context, name, cephCluster); getErr != nil {
				return errors.Wrapf(err, "failed to retrieve cephCluster %q in namespace %q to update csi cephx status. %v", name, name.Namespace)
			}
```